zebra-chain/src/orchard/tree.rs

```diff
@@ -228,6 +261,49 @@ impl NoteCommitmentTree {
     pub fn uncommitted() -> pallas::Base {
         pallas::Base::one().double()
     }
+
+    /// Count of note commitments added to the tree.
+    ///
+    /// For Orchard, the tree is capped at 2^32.
+    pub fn count(&self) -> u32 {
+        self.parents.iter().enumerate().fold(
+            match (self.left, self.right) {
+                (None, None) => 0,
+                (Some(_), None) => 1,
+                (Some(_), Some(_)) => 2,
+                (None, Some(_)) => unreachable!(),
```

Comment:
Missing documentation: Can you add a panic message explaining why this case is unreachable?

```suggestion
                (None, Some(_)) => unreachable!("roots are always added left to right"),
```